src/Analyzers/CSharp/CodeFixes/UseUTF8StringLiteral/UseUTF8StringLiteralCodeFixProvider.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.LanguageServices;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseUTF8StringLiteral
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.UseUTF8StringLiteral), Shared]
+    internal class UseUTF8StringLiteralCodeFixProvider : SyntaxEditorBasedCodeFixProvider
+    {
+        private const char QuoteCharacter = '"';
+        private const string Suffix = "u8";
+
+        [ImportingConstructor]
+        [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
+        public UseUTF8StringLiteralCodeFixProvider()
+        {
+        }
+
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+            ImmutableArray.Create(IDEDiagnosticIds.UseUTF8StringLiteralDiagnosticId);
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            context.RegisterCodeFix(CodeAction.Create(
+                CSharpAnalyzersResources.Use_UTF8_string_literal,
+                c => FixAsync(context.Document, context.Diagnostics[0], c),
+                nameof(CSharpAnalyzersResources.Use_UTF8_string_literal)),
+                context.Diagnostics);
+
+            return Task.CompletedTask;
+        }
+
+        protected override Task FixAllAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics,
+            SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            foreach (var diagnostic in diagnostics)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                var arrayNode = diagnostic.Location.FindNode(getInnermostNodeForTie: true, cancellationToken);
```

Comment:
Is it possible this ends up with a using statement syntax?

https://github.com/dotnet/roslyn/blob/0c7c0b33f0871fc4308eb2d75d77b87fc9293290/src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IUsingStatement.cs#L1189-L1194

Reply:
What the?!

Who writes weird language anyway

Reply:
> Who writes weird language anyway

Those are the kind of things one may **never** see or know about except from compiler tests 😄 

There are ZERO `Dispose` methods with a `params` parameter in the whole dotnet organization (per grep.app):

![image](https://user-images.githubusercontent.com/31348972/162554140-284caddb-37f2-4b1c-adc3-f5ebb2748d14.png)

Maybe skip the codefix for `IsImplicit` operations? Maybe even the analyzer (it will lead to false negatives, but for extremely rare cases)? Maybe do nothing and wait any user feedback (might never happen)?

Not sure what the best option is.

Reply:
So this isn't a problem because there is no initializer in the array creation expression, and unless I'm misunderstanding what this test is doing, there is no way to have anything there. Added a test just in case :)

Reply:
@davidwengier I see an `IArrayInitializerOperation` in the compiler test (it has zero elements, but it's not null). I was expecting `GetStringValue` to return an empty string and then we have a diagnostic. If the added test is passing, then I don't know why. But it's good!

Reply:
It has a null Type

Reply:
@davidwengier The `Type` check is done against `(arrayCreationExpression.Type as IArrayTypeSymbol)?.ElementType;`, which for the compiler test is `System.Object`:

https://github.com/dotnet/roslyn/blob/0c7c0b33f0871fc4308eb2d75d77b87fc9293290/src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IUsingStatement.cs#L1189

Did you replace `params object[]` in your test with `params byte[]`?

I could also be missing something.

Reply:
hmm, you're right. I did change the Param array type, but interestingly debugging through, for some reason my AnalyzeOperation method never gets called, which is why it passed.

At least we have a test that will fail if that ever changes.